sky/templates/gcp-ray.yml.j2

```
@@ -184,12 +232,25 @@ setup_commands:
 # items! The same comment applies for worker_start_ray_commands.
 #
 # Increment the following for catching performance bugs easier:
-#   current num items (num SSH connections): 1
+#   current num items (num SSH connections): 2
 head_start_ray_commands:
   # Start skylet daemon. (Should not place it in the head_setup_commands, otherwise it will run before sky is installed.)
   # NOTE: --disable-usage-stats in `ray start` saves 10 seconds of idle wait.
   # Line "which prlimit ..": increase the limit of the number of open files for the raylet process, as the `ulimit` may not take effect at this point, because it requires
   # all the sessions to be reloaded. This is a workaround.
+{%- if gpu is not none %}
+  - |
+    echo "Installing NVIDIA GPU driver." >> ~/.sky/nvlog
+    while ! nvidia-smi &> /dev/null
+    do
+      echo "Waiting for NVIDIA drivers to be installed..." >> ~/.sky/nvlog
+      sleep 5
+    done
+    # Magic number for waiting the nvidia driver to be ready and the instance
+    # to be rebooted.
+    sleep 18
```

```suggestion
    # Magic number for waiting for the nvidia driver to be ready and the instance
    # to be rebooted.
    sleep 18
```
By magic number, do you mean it's arbitrarily chosen?